ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py

```diff
@@ -63,75 +63,88 @@ def add_test(mapping: list, description: str) -> None:
 
 
 class AvdTestBGP(AvdTestBase):
-    """
-    AvdTestBGP class for BGP tests.
+    """AvdTestBGP class for BGP tests.
+
+    Supports IPv4, IPv6 and EVPN address families.
     """
 
     anta_module = "anta.tests.routing"
     categories = ["BGP"]
+    anta_tests = {}
+
+    def add_test(self, afi: str, bgp_neighbor_ip: str, safi: str | None = None) -> dict:
+        """Add a BGP test definition with the proper input parameters."""
+        formatted_afi = "IPv4" if afi.lower() == "ipv4" else "IPv6" if afi.lower() == "ipv6" else afi.upper()
+        formatted_safi = f" {safi.capitalize()}" if safi else ""
+        custom_field = f"BGP {formatted_afi}{formatted_safi} Peer: {bgp_neighbor_ip}"
+
+        address_family = {"afi": afi, "peers": [bgp_neighbor_ip]}
+        if safi:
+            address_family["safi"] = safi
+
+        self.anta_tests.setdefault(f"{self.anta_module}.bgp", []).append(
+            {
+                "VerifyBGPSpecificPeers": {
+                    "address_families": [address_family],
+                    "result_overwrite": {"categories": self.categories, "custom_field": custom_field},
+                },
+            },
+        )
+
+    def create_tests(self, afi: str, safi: str | None = None) -> None:
+        """Create BGP tests for the given AFI and SAFI."""
+        bgp_neighbors = get(self.structured_config, "router_bgp.neighbors", [])
+
+        # Retrieve peer groups and direct neighbors.
+        peer_groups = get(self.structured_config, f"router_bgp.address_family_{afi}.peer_groups", [])
+        direct_neighbors = get(self.structured_config, f"router_bgp.address_family_{afi}.neighbors", [])
+
+        # Only explicitely activated neighbors and peer groups are tested.
```

Comment:
```suggestion
        # Only explicitly activated neighbors and peer groups are tested.
```

Reply:
Fixed in 8a69d425679aa78680263e5e3429932f4d4a4adf